### ISSUE TITLE
Pandas is generating a sitecustomize error

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#2106]
_From [martijn....@gmail.com](https://code.google.com/u/105702808491802454567/) on 2015-01-07T10:26:11Z_

Spyder Version:  2.3.2
Python Version:  2.7.9
Qt Version    :  4.8.6, PyQt4 (API v2) 4.11.1 on Darwin
IPython >=0.13  :  2.3.1 (OK)
matplotlib >=1.0:  1.4.2 (OK)
pandas >=0.13.1 :  None (OK)
pep8 >=0.6      :  1.5.7 (OK)
psutil >=0.3    :  2.2.0 (OK)
pyflakes >=0.5.0:  0.8.1 (OK)
pygments >=1.6  :  2.0.1 (OK)
pylint >=0.25   :  1.4.0 (OK)
rope >=0.9.2    :  0.10.2 (OK)
sphinx >=0.6.6  :  1.2.3 (OK)
sympy >=0.7.0   :  0.7.6 (OK)
zmq >=2.1.11    :  14.4.1 (OK) 

What steps will reproduce the problem?

 1. Normal spyder startup
2. sitecustomize not executed
3. runfile functions not available 

What is the expected output? What do you see instead?

 Running python scripts using the built-in python interpreter.

On startup, the following is shown:

	'module' object has no attribute 'packaging'
	'import sitecustomize' failed; use -v for traceback
	Python 2.7.9 (default, Jan  7 2015, 12:39:27) 
	[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.56)] on darwin
	Type "help", "copyright", "credits" or "license" for more information.
	
	Imported NumPy 1.9.1, SciPy 0.14.0, Matplotlib 1.4.2
	
	Type "scientific" for more details.
	\>>> 

Please provide any additional information below

. I've checked the sitecustomize module, i.e. run inside spyder python terminal:

	\>>> from spyderlib.widgets.externalshell import sitecustomize
	cannot import name NaT
	\>>> cannot import name hashtable
	cannot import name hashtable

Doing a google search on these issues revealed the issue to be related to external dependency pandas (pandas==0.15.2).

Completely uninstalling pandas enabled using spyder as before. Installing an older version (pandas==0.13.1; the minimum recommend by my spyder install) also enabled the correct functioning again.

Note that this issue not that crucial to me as I do not really rely on the pandas library anyway.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=2106_

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2015-01-08T10:53:21Z_

Thanks for reporting. It probably is change in Pandas between 0.13.1 and 0.15.2. I'll try to fix it for our next release.

**Labels:** Cat-Console MS-v2.3.3

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2015-01-09T07:29:21Z_

I can't reproduce this error. We do import pandas in our sitecustomize to do a couple of monkey-patches, but that is protected with a try/except structure. Just to be sure, I tried to provoke an error in that part of the code and effectively couldn't :-)

However, I see you have another error before the sitecustomize one:

    'module' object has no attribute 'packaging'

which seems to point in another direction. I did a google search and found that that is related to setuptools, so maybe your setuptools is outdated or you have a problem with your installation.

To get the full sitecustomize error, could you run this command in a console?

SPYDER_DEBUG=3 spyder

**Summary:** Pandas sitecustomize error  \<span class="oldvalue"> (was: pandas sitecustomize issue) \</span>

**Comment by spyder-issues-migrator** [comment]
_From [martijn....@gmail.com](https://code.google.com/u/105702808491802454567/) on 2015-01-09T07:40:47Z_

Outputs: terminal (see `terminal.log`) and spyder python (`spyder.log`)

pip output for outdated packages:

	piano:~ mhemeryck$ pip list -o
	setuptools (Current: 9.1 Latest: 11.3.1)
	pip (Current: 6.0.3 Latest: 6.0.6)
	pandas (Current: 0.13.1 Latest: 0.15.2)
	scipy (Current: 0.14.0 Latest: 0.14.1)

pip and setuptools are indeed outdated, but I left them at this version because otherwise it seemed to conflict with my python (homebrew) install. Sorry if this issue is indeed not related to spyder. The pandas issue does not occur outside of my spyder install though.

**Attachment:** [terminal.log spyder.log](http://code.google.com/p/spyderlib/issues/detail?id=2106#hc3)

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2015-01-09T08:01:45Z_

I only see this possible error in spyder.log

    cannot import name hashtable

but doesn't seem serious because no traceback is generated.

I'm going to let this one open in case someone else reports the same problem :-)

**Labels:** -MS-v2.3.3 MS-v2.3.4

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2015-01-09T08:02:06Z_

**Summary:** Pandas is generating a sitecustomize error  \<span class="oldvalue"> (was: Pandas sitecustomize error) \</span>

**Comment by spyder-issues-migrator** [comment]
_From [martijn....@gmail.com](https://code.google.com/u/105702808491802454567/) on 2015-01-09T08:11:05Z_

I've just upgraded the outdated packages pip and setuptools and now everything seems to work fine. I've tried the same fix before, but then it didn't work.

Afterwards, I also did a brew untap homebrew/python, i.e. all my system-wide python packages are now managed by pip itself.

I've also updated the outdated pandas package (in short, everything updated to latest version).

Don't know what triggered it, sorry for the inconvenience.
